SkietbaanBE/SkietbaanBE/Controllers/UserController.cs

```diff
@@ -10,50 +12,71 @@ namespace SkietbaanBE.Controllers
     public class UserController : Controller
     {
         private ModelsContext _context;
-        public UserController(ModelsContext db) {
+        public UserController(ModelsContext db)
+        {
             _context = db;
         }
 
         //public UserController() { }
 
         // GET: api/User
         [HttpGet]
-        public IEnumerable<User> Get() {
+        public async Task<IEnumerable<User>> Get()
+        {
             var users = _context.Users.ToArray<User>();
             return users;
         }
 
         // GET: api/User/5
         [HttpGet("{id}", Name = "Get")]
-        public User Get(int id) {
+        public async Task<User> Get(int id)
+        {
             User user = _context.Users.Find(id);
             return user;
         }
 
-        [HttpGet("test/{ret}", Name = "Test")]
-        public string Test(string ret) {
-            return $"Returned : {ret}";
-        }
-
         // POST: api/User
         [HttpPost]
-        public virtual void Post([FromBody] User user) {
+        public async Task<ActionResult> Post([FromBody] User user)
+        {
+            //error handling, check if client provided valid data
+            if (user == null)
+            {
+                return new BadRequestObjectResult("user cannot be null");
+            }
+            else if (Get(user.Id) != null)
+            {
+                return new OkObjectResult("user already exists");
+            }
+
+            //get today's date and save it under user entry date
+            DateTime currentDate = new DateTime();
```

Comment:
Why declare a variable that is never realy used , simple asign to the user object in second line